server/routes/index.js

```diff
@@ -0,0 +1,152 @@
+/* eslint-disable no-console */
+
+const jwt = require('jsonwebtoken');
+const config = require('../config');
+
+const users = require('./user');
+const documents = require('./documents');
+const roles = require('./roles');
+
+// Require mongoose model
+const User = require('../models/user');
+
+module.exports = (apiRouter) => {
+  apiRouter.get('/', (req, res) => {
+    res.json({
+      message: 'Welcome to the DOCHERO api',
+    });
+  });
+
+// signup
+  apiRouter.route('/users')
+    .post((req, res) => {
+      const user = new User();
+
+      // Pass details of model instance to request
+      user.firstname = req.body.firstname;
+      user.lastname = req.body.lastname;
+      user.username = req.body.username;
+      user.email = req.body.email;
+      user.password = req.body.password;
+      user.title = req.body.title;
+
+      user.save((err) => {
+        if (err) {
+          if (err.code === 11000) {
+            res.json({
+              message: 'Please provide a unique username',
+            });
+          } else {
+            res.send(err);
+          }
+        } else {
+          // Generate the token on signup
+          const token = jwt.sign({
+            firstname: user.firstname,
+            username: user.username,
+            id: user._id,
+            title: user.title,
+          }, config.development.superSecret, {
+            // Token will expire in a day
+            expiresIn: 86400,
+          });
+          res.status(201).send({
+            message: 'User created successfully',
+            status: '201: Resource created',
+            user: user,
+            token: token,
+          });
+        }
+      });
+    });
+
+  // If they have the correct password, give them a token
+  apiRouter.post('/users/login', (req, res) => {
+    // Find the user
+    User.findOne({
+      username: req.body.username,
+    }).select('username password title').exec((err, user) => {
+      if (err) {
+        throw err;
+      }
+      // No user with that username was found
+      if (!user) {
+        res.status(404).send({
+          message: 'User not found',
+          status: '404: Resource Not Found',
+        });
+      }
+      // If a user with that username exists
+      else {
```

Comment:
## Critical

The else should start where the if block ends. You did that in earlier lines eg 42, but failed to do that here and a number of other places.